src/crepe/system/RenderSystem.cpp

```diff
@@ -24,13 +25,32 @@ void RenderSystem::update_camera() {
 		this->curr_cam = &cam;
 	}
 }
-void RenderSystem::render_sprites() const {
+
+bool sorting_comparison(const Sprite & a, const Sprite & b) {
+	if (a.sorting_in_layer > b.sorting_in_layer) return true;
+	if (a.sorting_in_layer == b.sorting_in_layer) return a.order_in_layer > b.order_in_layer;
+
+	return false;
+}
+
+std::vector<std::reference_wrapper<Sprite>>
+RenderSystem::sort(std::vector<std::reference_wrapper<Sprite>> & objs) {
+	if (objs.empty()) return {};
```

Comment:
Deze check is overbodig want er wordt sowieso al een lege vector gemaakt in dit geval (of dit `{}` of `sorted_objs` is maakt niet uit), en `std::sort` doet niks als `begin()` en `end()` de zelfde iterator zijn.